src/Nethermind/Nethermind.Optimism/OptimismEngineRpcModule.cs

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Nethermind.Consensus.Producers;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.JsonRpc;
+using Nethermind.JsonRpc.Modules;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin;
+using Nethermind.Merge.Plugin.Data;
+using Nethermind.Merge.Plugin.GC;
+using Nethermind.Merge.Plugin.Handlers;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.Optimism;
+
+public class OptimismEngineRpcModule : IOptimismEngineRpcModule
+{
+    private readonly IEngineRpcModule _engineRpcModule;
+
+    public async Task<ResultWrapper<ForkchoiceUpdatedV1Result>> engine_forkchoiceUpdatedV1(ForkchoiceStateV1 forkchoiceState, OptimismPayloadAttributes? payloadAttributes = null)
+    {
+        if (payloadAttributes is { GasLimit: 0 })
+        {
+            return ForkchoiceUpdatedV1Result.Error("GasLimit is required", MergeErrorCodes.InvalidForkchoiceState);
+        }
```

Comment:
Question, on normal Ethereum now we first apply FCU and only then we Validate the payloadAttributes and if it fails, we don't build the block. Check: https://github.com/ethereum/execution-apis/pull/476

So to achieve it here, we would have to make `PayloadAttributesExtensions` methods like `GetVersion` and `Validate` actually virtual on the `PayloadAttributes` and validate with the same code path as we validate in normal FCU.

Unless Optimism is different here.